src/query.rs

```diff
@@ -289,6 +337,43 @@ pub fn q_once<'sqlite, 'schema, 'query, T>
     run_algebrized_query(sqlite, algebrized)
 }
 
+pub fn q_prepare<'conn, 'sqlite, 'schema, 'query, T>
+(conn: &'conn Conn,
```

Comment:
You don't actually use the `conn` here except to push it into the output. And it's not actually used by the output except to check the current schema. So my suspicion is that you shouldn't take `conn` as an argument: you might add a concept of invalidation as a wrapper around this, but it's not an essential part of a prepared statement — e.g., prepare and use within an `InProgress` or `InProgressRead` doesn't need invalidation checks.

Reply:
Oh, that makes sense. So, similar to how `InProgressRead` is a wrapper around `InProgress` now, we'd have a `LongLivedPreparedQuery` or something wrap a `PreparedQuery` and ensure the schema hasn't changed?

Reply:
Yes, or an inversion of that where a `PreparedQuery` (or a related type) is actually _owned_ by a `Conn` or similar (`QueryCache`?), and you can only _borrow_ it:

```
let prepared: &PreparedQuery = conn.prepare_query("…")?;
prepared.run(…);
…
let prepared = conn.prepare_query("…")?;    // Gives us a borrow to the same statement.
…
```